From d12f711f0c080affa859d42ba0a1663e23549379 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@freescale.com>
Date: Fri, 7 Aug 2015 22:20:25 +0800
Subject: [PATCH 0099/5242] MLK-10282-3 ARM: imx7d-12x12-arm2: enable usbotg1
 and usbotg2

commit  7bcf4c237729cd106fa6a48d46fb53eee34ab3dc from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable usbotg1 and usbotg2

Signed-off-by: Peter Chen <peter.chen@freescale.com>
Fixed pinctrl during 4.14 rebase for upstream LPSR rename.
Also it should be _GPIO1_IO05__GPIO1_IO5 not _GPIO1_IO05__GPIO5_IO5.
Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts |   44 +++++++++++++++++++++++++
 1 file changed, 44 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
index 6038d57..f2c0c61 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
@@ -24,6 +24,8 @@
 
 	regulators {
 		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <0>;
 
 		reg_sd1_vmmc: sd1_vmmc{
 			compatible = "regulator-fixed";
@@ -42,6 +44,26 @@
 			gpio = <&gpio5 11 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
+
+		reg_usb_otg1_vbus: regulator@0 {
+			compatible = "regulator-fixed";
+			reg = <0>;
+			regulator-name = "usb_otg1_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio1 5 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
+
+		reg_usb_otg2_vbus: regulator@1 {
+			compatible = "regulator-fixed";
+			reg = <1>;
+			regulator-name = "usb_otg2_vbus";
+			regulator-min-microvolt = <5000000>;
+			regulator-max-microvolt = <5000000>;
+			gpio = <&gpio1 7 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
 	};
 
 	memory {
@@ -344,6 +366,18 @@
 			>;
 		};
 
+		pinctrl_usbotg1_vbus: usbotg1vbusgrp {
+			fsl,pins = <
+				MX7D_PAD_LPSR_GPIO1_IO05__GPIO1_IO5    0x14
+			>;
+		};
+
+		pinctrl_usbotg2_vbus: usbotg2vbusgrp {
+			fsl,pins = <
+				MX7D_PAD_LPSR_GPIO1_IO07__GPIO1_IO7    0x14
+			>;
+		};
+
 		pinctrl_usdhc1_1: usdhc1grp-1 {
 			fsl,pins = <
 				MX7D_PAD_SD1_CMD__SD1_CMD     0x17059
@@ -449,6 +483,16 @@
 	status = "okay";
 };
 
+&usbotg1 {
+	vbus-supply = <&reg_usb_otg1_vbus>;
+	status = "okay";
+};
+
+&usbotg2 {
+	vbus-supply = <&reg_usb_otg2_vbus>;
+	status = "okay";
+};
+
 &usdhc1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc1_1>;
-- 
1.7.9.5

